homeassistant/components/emoncms_history/__init__.py

```diff
@@ -42,61 +43,51 @@
+    hass: HomeAssistant,
+    emoncms_client: EmoncmsClient,
+    whitelist: list[str],
+    node: str | int,
+    _: datetime,
+) -> None:
+    """Send data to Emoncms."""
+    payload_dict = {}
+
+    for entity_id in whitelist:
+        state = hass.states.get(entity_id)
+        if state is None or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE):
+            continue
         try:
-            fullurl = f"{url}/input/post.json"
-            data = {"apikey": apikey, "data": payload}
-            parameters = {"node": node}
-            req = requests.post(
-                fullurl, params=parameters, data=data, allow_redirects=True, timeout=5
-            )
-
-        except requests.exceptions.RequestException:
-            _LOGGER.error("Error saving data '%s' to '%s'", payload, fullurl)
+            payload_dict[entity_id] = state_helper.state_as_number(state)
+        except ValueError:
+            continue
 
+    if payload_dict:
+        try:
+            await emoncms_client.async_input_post(data=payload_dict, node=node)
+        except (aiohttp.ClientError, TimeoutError) as err:
+            _LOGGER.warning("Network error when sending data to Emoncms: %s", err)
+        except ValueError as err:
+            _LOGGER.warning("Value error when preparing data for Emoncms: %s", err)
         else:
-            if req.status_code != HTTPStatus.OK:
-                _LOGGER.error(
-                    "Error saving data %s to %s (http status code = %d)",
-                    payload,
-                    fullurl,
-                    req.status_code,
-                )
-
-    def update_emoncms(time):
-        """Send whitelisted entities states regularly to Emoncms."""
-        payload_dict = {}
-
-        for entity_id in whitelist:
-            state = hass.states.get(entity_id)
+            _LOGGER.debug("Sent data to Emoncms: %s", payload_dict)
 
-            if state is None or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE):
-                continue
 
-            try:
-                payload_dict[entity_id] = state_helper.state_as_number(state)
-            except ValueError:
-                continue
-
-        if payload_dict:
-            payload = ",".join(f"{key}:{val}" for key, val in payload_dict.items())
-
-            send_data(
-                conf.get(CONF_URL),
-                conf.get(CONF_API_KEY),
-                str(conf.get(CONF_INPUTNODE)),
-                f"{{{payload}}}",
-            )
-
-        track_point_in_time(
-            hass, update_emoncms, time + timedelta(seconds=conf.get(CONF_SCAN_INTERVAL))
-        )
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Emoncms history component."""
+    conf = config[DOMAIN]
+    whitelist = conf.get(CONF_WHITELIST)
+    input_node = str(conf.get(CONF_INPUTNODE))
+
+    emoncms_client = EmoncmsClient(
+        url=conf.get(CONF_URL),
+        api_key=conf.get(CONF_API_KEY),
```

Comment:
For a future PR, the values are always there, so we don't need to do `.get` right?

I am fine with keeping this for now as the old code already did this